manyfews/calculations/zentra.py

```diff
@@ -0,0 +1,238 @@
+                "value"
+            ]
+        )  # vapor pressure, 'units': ' kPa'
+
+        atmosPressure.append(
+            zentraData["device"]["timeseries"][0]["configuration"]["values"][i][3][9][
+                "value"
+            ]
+        )  # Atmospheric Pressure, 'units': 'kPa'
+
+        VPD.append(
+            zentraData["device"]["timeseries"][0]["configuration"]["values"][i][3][14][
+                "value"
+            ]
+        )  # VPD, 'units': ' kPa'
+
+        rhTemp.append(
+            zentraData["device"]["timeseries"][0]["configuration"]["values"][i][3][13][
+                "value"
+            ]
+        )  # RH Senor Temp, 'units': ' °C'
+
+        gustSpeed.append(
+            zentraData["device"]["timeseries"][0]["configuration"]["values"][i][3][6][
+                "value"
+            ]
+        )  # Gust Speed, 'units': ' m/s'
+
+        maxPrecipRate.append(
+            zentraData["device"]["timeseries"][0]["configuration"]["values"][i][3][12][
+                "value"
+            ]
+        )  # Max Precip Rate, 'units': ' mm/h'
+
+        # convert time stamp to Date
+        ts = zentraData["device"]["timeseries"][0]["configuration"]["values"][i][
+            0
+        ]  # time stamp
+        tz = datetime.timezone(datetime.timedelta(hours=6))  # time zone
+        date = datetime.datetime.fromtimestamp(ts, tz)  # convert date
+        covertDate.append(date)
+
+        # calculate RH by: esTair = 0.611*EXP((17.502*Tc)/(240.97+Tc))
+        #                  RH = VP / esTair
+        #                 which:Tc is the Air Temperature
+        #                       VP is the Vapour Pressure
+        #                       RH is the Relative Humidity between zero and one.
+
+        Tempair = zentraData["device"]["timeseries"][0]["configuration"]["values"][i][
+            3
+        ][7]["value"]
+
+        vapPressure = zentraData["device"]["timeseries"][0]["configuration"]["values"][
+            i
+        ][3][8]["value"]
+        rh = vapPressure / (0.611 * (math.exp((17.502 * Tempair) / (240.97 + Tempair))))
+
+        if rh > 1:
+            rh = 1
+        elif rh < 0:
+            rh = 0
+
+        RH.append(rh)
+
+    # output data sets into a dictionary
+    zentraDataSum = [
+        covertDate,
+        tStamp,
+        solar,
+        precip,
+        wDirection,
+        wSpeed,
+        gustSpeed,
+        vaporPressure,
+        atmosPressure,
+        airTem,
+        maxPrecipRate,
+        rhTemp,
+        VPD,
+        RH,
+    ]
+    zentraDataKeys = [
+        "local date",
+        "time stamp",
+        "solar radiation",
+        "precipitation",
+        "wind direction",
+        "wind speed",
+        "guest speed",
+        "vapor pressure",
+        "atmospheric pressure",
+        "air temperature",
+        "max precipitation Rate",
+        "RH senor temperature",
+        "VPD",
+        "Relative Humidity",
+    ]
```

Comment:
I'm not keen on having hard-coded names here. Could we just create ZentraReading objects at this point and return a list of objects?

Reply:
accepted and changed.